LLMonFHIR/FHIR Display/FHIRMultipleResourceInterpreter.swift

```diff
@@ -0,0 +1,88 @@
+//
+// This source file is part of the Stanford LLM on FHIR project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import OpenAI
+import Spezi
+import SpeziLocalStorage
+import SpeziOpenAI
+import SwiftUI
+
+
+private enum FHIRMultipleResourceInterpreterConstants {
+    static let storageKey = "FHIRMultipleResourceInterpreter.Cache"
+}
+
+
+class FHIRMultipleResourceInterpreter<ComponentStandard: Standard>: DefaultInitializable, Component, ObservableObject, ObservableObjectProvider {
+    typealias MultipleResourceInterpretation = String
+    @Dependency private var localStorage: LocalStorage
+    @Dependency private var openAIComponent = OpenAIComponent()
+    
+    
+    var interpretation: String? {
+        willSet {
+            Task { @MainActor in
+                objectWillChange.send()
+            }
+        }
+        didSet {
+            do {
+                try localStorage.store(interpretation, storageKey: FHIRMultipleResourceInterpreterConstants.storageKey)
+            } catch {
+                print(error)
+            }
+        }
+    }
+    
+    
+    required init() {}
+    
+    
+    func configure() {
+        guard let cachedInterpretation: MultipleResourceInterpretation = try? localStorage.read(
+            storageKey: FHIRMultipleResourceInterpreterConstants.storageKey
+        ) else {
+            return
+        }
+        self.interpretation = cachedInterpretation
+    }
+    
+    func interpretMultipleResources(resources: [FHIRResource]) async throws {
+        guard interpretation != nil else {
+            let chatStreamResults = try await openAIComponent.queryAPI(withChat: [systemPrompt(forResources: resources)])
+            
+            for try await chatStreamResult in chatStreamResults {
+                for choice in chatStreamResult.choices {
+                    let previousInterpretation = interpretation ?? ""
+                    interpretation = (interpretation ?? "") + (choice.delta.content ?? "")
+                }
+            }
+            return
+        }
```

Comment:
A general code suggestion that is also applicable beyond this change: A general thing that we aim for in Swift code is keeping the "golden" path (the one where we are in a non-error state) outside of nested statements. This allows you to make the code a bit more readable and guard statements are a perfect way to enforce that we exit early in case if an error.

Therefore I would suggest to change the code like this:
```suggestion
        guard interpretation == nil else {
            return
        }
        
        let chatStreamResults = try await openAIComponent.queryAPI(withChat: [systemPrompt(forResources: resources)])
        
        for try await chatStreamResult in chatStreamResults {
            for choice in chatStreamResult.choices {
                let previousInterpretation = interpretation ?? ""
                interpretation = (interpretation ?? "") + (choice.delta.content ?? "")
            }
        }
```